rulesets/google-enhanced-ecommerce-ga4.js

```diff
@@ -0,0 +1,167 @@
+window['_dlo_rules_google_ec_ga4'] = [
+  {
+    "id": "fs-ga4-e-commerce-select_item",
+    "source": "dataLayer",
+    "operators": [
+      { "name": "query", "select": "$[?(event=select_item)]" },
+      { "name": "query", "select": "$.ecommerce.items[0]" },
```

Comment:
Could `ecommerce.items` ever be empty? Is this a test case we should capture?

Reply:
I think we should.  Two outcomes, either 1) the query operator causes the chain to exit because the selected object is undefined or 2) we could add an operator step before to check that the index 0's value is not undefined.

But I have a feeling the isEmptyData check will cause the chain to exit.
```
// if the data is null or empty, it is a signal to stop processing
        // this can happen if an upstream handler needed to prevent a downstream operator
        // and we don't want to submit empty events to the configured destination
        if (handledData === null || DataHandler.isEmptyData(handledData)) {
          this.runDebugger(`[${i}] ${name} halted`, handledData, '  ');
          return null;
        }
```

Reply:
It looks like we gracefully handle empty `items` arrays. I added tests to verify all events except `purchase` result in:

- 0 events
- 0 errors